hazelcast-documentation/src/RingBuffer.md

```diff
@@ -1,49 +1,236 @@
+* startSequence: the sequence of the first item to read
+* minCount: the minimum number of items to read. If you don't want to block, provide 0. If you do want to block for at least one item,
+provide 1.
+* the maximum number of items to retrieve. There is a hard cap on the maxCount and that is 1000.
+* filter: a function that accept an item and checks if it should be returned. If no filtering should be applied, pass null.
+
+A full example:
+```java
+long sequence = rb.headSequence();
+for(;;) {
+    ICompletableFuture<ReadResultSet<String>> f = rb.readManyAsync(sequence, 1, 10, null);
+    ReadResultSet<String> rs = f.get();
+    for (String s : rs) {
+        System.out.println(s);
+    }
+    sequence+=rs.readCount();
+}
+```        
+Please take a careful look at how the sequence is being incremented. You can't always rely on the number of items being returned
+if items are filtered out.
+
+### Async methods
+The Ringbuffer provides asynchronous methods for the more powerful methods like batched reading with filtering or batch writing. 
+To make these methods synchronous, just call `get()` on the returned future.
+
+Example:
+```java
+ICompletableFuture f = ringbuffer.addAsync(item, OverflowPolicy.FAIL);
+f.get();
 ```
 
+But the ICompletable future can also be used to get notified when the operation has completed. For example when you want to 
+get notified when a batch of reads has completed:
+```java
+ICompletableFuture<ReadResultSet<String>> f = rb.readManyAsync(sequence, min, max, someFilter);
+f.andThen(new ExecutionCallback<ReadResultSet<String>>() {
+   @Override
+   public void onResponse(ReadResultSet<String> response) {
+        for (String s : response) {
+            System.out.println("Received:" + s);
+        }
+   }
+
+   @Override
+   public void onFailure(Throwable t) {
+        t.printStackTrace();
+   }
+});
+```
+THe advantage of this approach is that the thread that does the call isn't block till the response is returned.
 
-The RingBuffer can be configured with a time to live in seconds. Using this setting, you can control how long the items remain in the RingBuffer before they are deleted. By default, the time to live is set to 0, meaning that unless the item is overwritten, it will remain in the RingBuffer indefinitely. If a time to live is set and an item is added, then depending on the OverwritePolicy, either the oldest item is overwritten, or the call is rejected. 
+### Full Configuration examples
 
-The RingBuffer can also be configured with an InMemoryFormat which controls the format of stored items. By default, `BINARY` is used, meaning that the object is stored in a serialized form. You can select the `OBJECT` InMemoryFormat, which is useful when filtering is applied or when the `OBJECT` InMemoryFormat can lead to a smaller memory footprint than a `BINARY`. 
+A Ringbuffer can be configured in the XML. Below a full example of a ringbuffer called 'rb' can be found. It is configured
```

Comment:
The second sentence should more be like "The following snippet shows the configuration of a ringbuffer called 'rb'. The configuration is modeled after the ringbuffers defaults.